### DISCUSSION TITLE
How to add an .aar/.jar dependency to soot classpath through Options?

### DISCUSSION THREAD

**Discussion by joaoluizn** [soot-oss/soot#1512] [Q&A]
I've tried to disable allow phantom refs when trying to analyze an Android APK, but it says:

```couldn't find class: androidx.paging.PositionalDataSource$LoadInitialCallback (is your soot-class-path set properly?)```

So, I started to search by this dependency, found some ```androidx.paging``` on Maven central, but they are all ```.aar``` files.

Tried to load one into the classpath using ```Options.v().set_soot_class_path("<path_to_lib>/lib.aar")```, but it isn't working properly.

Any suggestion on what can I do?

Thanks!

**Comment by MarcMil** [comment]
If you take a look at an AAR file, you will notice that it is a ZIP file. That ZIP file contains a file called classes.jar, which contains the actual code. AFAIK, soot cannot handle AAR files at the moment, but it surely can ingest the jar file. As such, extract the AAR file and put the classes.jar file on the class path.
Besides, you might want to enable --allow-phantom-refs. Android apps in particular seem to have some references with no actual implementations in some places. Also, be sure you enable multidex via --process-multiple-dex. When the code for that class happens to be in a different dex file in multi dex apks, this might also explain the problem.
